app/services/analytics.rb

```diff
@@ -134,13 +134,22 @@ def sp_request_attributes
       [v.name.sub('http://idmanagement.gov/ns/assurance/', ''), true]
     end.to_h
     attributes.reject! { |_key, value| value == false }
+    attributes.merge!(differentiator)
     attributes.transform_keys! do |key|
       key.to_s.chomp('?').to_sym
     end
 
     { sp_request: attributes }
   end
 
+  def differentiator
```

Comment:
if this is effectively only merging a hash with a single consistent key if there is a value, could we assign in `sp_request_attributes` above if there's a value? Maybe it's worth memoizing the value in this case, but something like

```ruby
    if differentiator.present?
      attributes[:app_differentiator] = differentiator
    end
```

Reply:
I agree with ☝️

Reply:
I also think that "differentiator" is hard to say, and that "identifier" is easier 😄

Reply:
(But would not hold this up over bikeshedding the name any more)

Reply:
updated in https://github.com/18F/identity-idp/pull/11262/commits/3472c1c7270c4c3bae14ef3a1ace61986462de60

matt, i agree with you (also i misspell it every time), but since we have the `app_id` field on service providers, this feels more easily understandable as not connected to that value.